client/public/static/js/verification.js

```diff
@@ -7,55 +8,48 @@ $(document).ready(function () {
     console.log("Verification Code from SessionStorage:", utils.getFromStorage("verificationCode"));
   }
 
-  // Handles verification form submission logic
-  $("#verifyAccountForm").on("submit", function (event) {
-    event.preventDefault();
-    utils.clearMessageFromAuthModal(authClassesToRemove);
+ // Handles verification form submission logic
+ $("#verifyAccountForm").on("submit", function (event) {
+  event.preventDefault();
+  const verificationCode = $("#verificationCodeInput").val();
+  const username = utils.getFromStorage("username");
 
-    const form = $(this);
-    const formArray = form.serializeArray();
+  if (!username || !verificationCode) {
+    $("#verifyAccountForm").prepend(
+      `<div class="alert alert-danger">Username or verification code missing.</div>`
+    );
+    return;
+  }
 
-    const [verificationCode] = formArray.map(({ value }) => value);
-    const username = utils.getFromStorage("username");
+  fetch("/verify-account", {
+    method: "PUT",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({ username, verificationCode }),
+  })
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error("Verification failed.");
+      }
+      return response.json();
+    })
+    .then((data) => {
+      // Store auth token and redirect to dashboard
+      utils.setStorage("authToken", data.authToken);
+      utils.setStorage("loginMessage", "Account verified and logged in successfully.");
+      window.location.href = "/dashboard";
```

Comment:
Not sure where you got this from, there is no "/dashboard". You can just redirect to home "/".